File: LocationinRaidfinder.java
package locationinRAID;

import org.apache.commons.lang3.StringUtils;

public class LocationinRaidfinder {

	/**
	 * @param args
	 */
	
	
String initialPath;	
String[] fields;

//fields for fileSystemLoader and dicomPathBuild
String IIDGroup;
boolean ValidMidsector;
String MidSector;
String MidLabel;
String MidInstitution;
String IIDdt;
String IID_tm;
String YYYY;
String MM;
String DD;

	
public String pathFinder(String path, boolean gDcm) {
	
	String finalPath;
	
	if (gDcm = false){
		finalPath = "Sorry, have not yet written for non DICOM";
	}
	else{
	
		initialPath = path;	
		fields = startparse(initialPath);
		fileSystemLoader(fields);
		finalPath = dicomPathBuild();
	}
		
return finalPath;	

}

public String[] startparse(String row){
	
	String delims = "[ ]+";
	String [] a = row.split(delims);
	return a;
	
}

public void fileSystemLoader(String[] qLoad) {
	
	int t = 0;
	boolean cmp = false;
	int IIDGroupCount = 0;
	while (cmp == false){
		int count = StringUtils.countMatches(qLoad[t], ".");
		//finding IIDgroup first for reference
		
		if (count > 5 && t > 2 && t < (qLoad.length -2)){
			IIDGroup = qLoad[t];
			IIDGroupCount = t;
			System.out.println("IIDGroup is:" + IIDGroup);}
		
		if (t == qLoad.length - 2){
			if (qLoad[t].contains("invalid")){
				ValidMidsector = false;}
				else{ 
				ValidMidsector = true;
				MidSector = qLoad[t];}}
		
		if (t == qLoad.length - 3){
			MidLabel = qLoad[t];}
		
		if (t == qLoad.length -4){
			MidInstitution = qLoad[t];}
		
		if (IIDGroup != null && t == IIDGroupCount - 4){
			IIDdt = qLoad[t];
			IIDdt = IIDdt.replaceAll("-", ".");
			String gdelims = "[.]+";
			String[] c = IIDdt.split(gdelims);
			YYYY = c[0];
			MM = c[1];
			DD = c[2];}
		
		if (IIDdt != null && t == IIDGroupCount - 3){
			IID_tm = qLoad[t];
			IID_tm = IID_tm.replaceAll(":", ".");
			cmp = true;}
		
		if (t < qLoad.length - 1){
			t++;}
		else{
			t = 1;}
	}
	
	
	
}

public String dicomPathBuild() {
	
	String m = null;
	String j = null;
	
	if (ValidMidsector = true){
		m = "ls /MOUNT/" + MidLabel + "/" + MidInstitution + "/" + YYYY + MM + "/" + DD + "/" + MidSector + "/" + IIDGroup + "/" + IID_tm + "--" + IIDdt + "*";
		return m;}
	else 
		j = "ls /MOUNT/" + MidLabel + "/" + MidInstitution + "/" + YYYY + MM + "/" + DD + "/" + IIDGroup + "/" + IID_tm + "--" + IIDdt + "*";
		return j;
	
}



}

File: locationinRaidFinder.py
'''
Created on Mar 1, 2016

@author: Anthony

Parsing query string data and returning a 
file path from the result
'''

IIDGroup=None
MidSector=None
MidLabel=None
MidInstitution=None
IIDdt=None
IID_tm=None
YYYY=None
MM=None
DD=None


def pathFinder(path, gDcm):
    
    finalPath = ""
    
    if gDcm == False:
        finalPath = "Sorry, have not yet written for non DICOM"
    else:
        initialPath = path
        fields = startparse(initialPath)
        finalPath = fileSystemLoader(fields)
        
    return finalPath

def startparse(row):
    a = row.split()
    return a;
    
def fileSystemLoader(qLoad):
    global IIDGroup, MidSector, MidLabel, MidInstitution, IIDdt, IID_tm, YYYY, MM, DD
    t = 0
    cmplt = False
    fsLoaderCount = 0
    IIDGroupCount = 0
    MisErr = False
    ValidMidsector = True
    while cmplt == False:
        #print "t= " + qLoad[t],
        count = qLoad[t].count(".")  #finding IIDgroup first for reference
        #print "count= " + str(count)
        #if t >= 13:
        #    break
        
        if count > 5 and t > 2 and t < (len(qLoad) - 2):
            IIDGroup = qLoad[t]
            IIDGroupCount = t
            #print"IIDGroup is: " + IIDGroup
            
        if (t == len(qLoad) - 2):
            if qLoad[t].find("invalid") != -1:
                ValidMidsector = False
            else:
                ValidMidsector = True
                MidSector = qLoad[t]
               
                
        if t == len(qLoad) - 3:
            MidLabel = qLoad[t]
            
        if t == len(qLoad) - 4:
            MidInstitution = qLoad[t]
            
        if IIDGroup != None and t == IIDGroupCount - 4:
            IIDdt = qLoad[t]
            IIDdt = IIDdt.replace("-", ".")
            c = IIDdt.split(".")
            YYYY = c[0]
            MM = c[1]
            DD = c[2]
            
        if IIDdt != None and t == IIDGroupCount - 3:
            IID_tm = qLoad[t]
            IID_tm = IID_tm.replace(":", ".")
            cmplt = True
            #print "IID_tm: " + str(IID_tm)
             
        if t < len(qLoad) - 1:
            t+=1
        else:
            t = 1
        
        fsLoaderCount+=1    
        if fsLoaderCount >= 30:
            MisErr = True
            break     
        
    dcmPath = dicomPathBuild(ValidMidsector, MisErr)
    return dcmPath
            
def dicomPathBuild(ValidMidsector, MisErr):
    m = None
    j = None 
    
    if MisErr == True:
        return "The File System Loader does not recognize the text you copied"
    
    elif ValidMidsector == True:
        m = "/MOUNT/" + MidLabel + "/" + MidInstitution + "/" + str(YYYY) + str(MM) + "/" + str(DD) + "/" + MidSector + "/" + IIDGroup + "/" + str(IID_tm) + "--" + str(IIDdt) + "*"            
        return m
    else:
        j = "/MOUNT/" + MidLabel + "/" + MidInstitution + "/" + str(YYYY) + str(MM) + "/" + str(DD) + "/" + IIDGroup + "/" + str(IID_tm) + "--" + str(IIDdt) + "*"
        return j           
                    
                            
            
        
